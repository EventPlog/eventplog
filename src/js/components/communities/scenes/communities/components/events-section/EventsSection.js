import React from 'react'
import { Link } from 'react-router-dom'

// internal
import Sidebar from 'js/components/shared/sidebar'
import Loading from 'js/components/shared/loading'
import { genEventLink, genCommunityLink } from 'js/utils'

export const generateTitle = (event = {}, community = {}) => (
  <Link to={genEventLink(event, community)}>
    {event.title}
  </Link>
)

export const generateDescription = (community) => (
  <span>
    By <Link to={genCommunityLink(community)}>
      {community.name}
    </Link>
  </span>
)

export const generateMeta = (event) => (
  `${event.interested_persons} people interested`
)


const EventsSection = ({ events }) => {
  const {loading, error, data } = events
  return (
    <Sidebar title="Events you may like">
      {loading && <Loading />}
      {error && <Loading.Error msg={events.error} />}
      {(!loading && !error && data) && data.map(({community, description: d, featured_image, ...event}) => {
          const title = generateTitle(event, community);
          const description = community ? generateDescription(community) : '';
          const meta = generateMeta(event)
          return <Sidebar.Card
                    key={event.id}
                    {...{title, description, featured_image, meta}} />
        }
      )}
    </Sidebar>
  )
}

export default EventsSection