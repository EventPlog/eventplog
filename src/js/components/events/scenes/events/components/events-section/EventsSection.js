import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'

// internal
import ContentPanel from 'js/components/shared/content-panel'
import Loading from 'js/components/shared/loading'
import Error from 'js/components/shared/loading/Error'
import Button from 'js/components/shared/button'
import Pagination from 'js/components/shared/pagination'
import { pluralize } from 'js/utils'

export const generateTitle = (event = {}) => {
  const community = event.community || {}
  return (
    <Link to={`/communities/${event.community_id}/events/${event.id}`}>
      {event.title}
    </Link>
  )
}

export const generateDescription = (community = {}) => (
  <span>
    By <Link to={`/communities/${community.id}`}>
      {community.name}
    </Link>
  </span>
)

export const generateMeta = (event = {}) => ([
  <ul key={`date${event.id}`}>
    <li>
      {event.date}
    </li>
    <li>
      {event.time}
    </li>
    <li>
      {event.venue}
    </li>
  </ul>,
  <ul key={`interest${event.id}`}>
    <li>
      {event.interested_persons} {pluralize('person', event.interested_persons)} interested
    </li>
  </ul>
])

export const generateCTA = (handleClick) => (
  <Button onClick={handleClick}>
    Interested
  </Button>
)

const EventsSection = ({
  title,
  events = {},
  attendEvent,
  getEvents,
}) => {
  const {loading, error, data = [], meta = {}} = events;
  const shouldDisplayData = (!loading && !error && data);
  return (
    <ContentPanel title={title}>
      {loading && <Loading />}
      {error && <Loading.Error msg={events.error} />}
      {shouldDisplayData && data.map(({featured_image, ...event}) => {
          const title = generateTitle(event)
          const description = generateDescription(event.community)
          const meta = generateMeta(event)
          const btn = {onClick: () => attendEvent(event), text: 'interested'}
          const titleLink = `/communities/${event.community_id}/events/${event.id}`

          return (
            <ContentPanel.Card
              key={event.id}
              {...{title, description, featured_image, meta, btn, titleLink}}
              showButton={!event.is_attending} />
          )
        }
      )}
      {shouldDisplayData && data.length < 1 && <p>No events to display today ...</p>}
      {
        meta && meta.total_pages && (data.length > 0 || meta.current_page > 1)
          ? <Pagination totalPages={meta.total_pages}
                        activePage={meta.current_page}
                        onPageChange={getEvents} />
          : ''
      }
    </ContentPanel>
  )
}

export default EventsSection