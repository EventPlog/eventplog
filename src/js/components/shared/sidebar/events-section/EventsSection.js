import React from 'react'
import { Link } from 'react-router-dom'

// internal
import Sidebar from 'js/components/shared/sidebar'
import Loading from 'js/components/shared/loading'

export const generateTitle = (event = {}) => (
  <Link to={`/communities/${event.community_id}/events/${event.id}`}>
    {event.title}
  </Link>
)

export const generateDescription = (community) => (
  <span>
    By <Link to={`/communities/${community.id}`}>
      {community.name}
    </Link>
  </span>
)

export const generateMeta = (event) => (
  `${event.interested_persons} people interested`
)


const EventsSection = ({
  title,
  events = {data: [], meta: {}},
  attendEvent
}) => {
  const {loading, error, data = {} } = events
  return (
    <Sidebar title={title || "Events you may like"}>
      {loading && <Loading />}
      {error && <Loading.Error msg={events.error} />}
      {(!loading && !error && data) && data.map(({community, description: d, featured_image, ...event}) => {
          const title = generateTitle(event);
          const description = generateDescription(community);
          const meta = generateMeta(event)
          const btn = event.is_attending
                      ? {}
                      : {onClick: () => attendEvent(event), text: 'Interested'}
          return (
            <Link to={`/communities/${event.community_id}/events/${event.id}`}>
              <Sidebar.Card key={event.id}
                             {...{title, description,
                             featured_image, meta, btn}} />
            </Link>
          )
        }
      )}
    </Sidebar>
  )
}

export default EventsSection