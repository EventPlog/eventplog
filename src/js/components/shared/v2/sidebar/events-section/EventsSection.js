import React from 'react'
import { Link } from 'react-router-dom'

// internal
import Sidebar from 'js/components/shared/sidebar'
import Loading from 'js/components/shared/loading'
import { pluralize, genCommunityLink, genEventLink } from 'js/utils'

export const generateTitle = (event = {}, community = {}) => (
  <Link to={genEventLink(event, community)}>
    {event.title}
  </Link>
)

export const generateDescription = (community = {}) => (
  <span>
    By <Link to={genCommunityLink(community)}>
      {community.name}
    </Link>
  </span>
)

export const generateMeta = (event) => (
  `${event.interested_persons} ${pluralize('person', event.interested_persons)} interested`
)


const EventsSection = ({
  title,
  events = {data: [], meta: {}},
  attendEvent
}) => {
  const {loading, error, data = [] } = events
  return (
    <Sidebar title={title || "Events you may like"}>
      {loading && <Loading />}
      {error && <Loading.Error msg={events.error} />}
      {(!loading && !error && data) && data.map(({community, description: d, featured_image, ...event}) => {
          const title = generateTitle(event, community);
          const description = community ? generateDescription(community) : '';
          const meta = generateMeta(event)
          const titleLink = genEventLink(event, community)
          const btn = event.is_attending
                      ? {}
                      : {onClick: () => attendEvent(event), text: 'Interested'}
          return (
            <Sidebar.Card key={event.id}
                           {...{title, description, titleLink,
                           featured_image, meta, btn}} />
          )
        }
      )}
    </Sidebar>
  )
}

export default EventsSection