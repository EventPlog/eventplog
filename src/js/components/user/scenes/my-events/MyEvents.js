import React, { Component } from 'react'
import { Grid, Menu, Segment } from 'semantic-ui-react'
import styled from 'styled-components'

// =========== INTERNAL ============
import EventsSection from 'js/components/events/scenes/events/components/events-section'
import { media } from 'js/styles/mixins'
import sharedStyles from '../sharedStyles'

const StyledEvents = styled.div`
  ${sharedStyles}
`

const MyEvents = ({
  user,
  labelVerbMapping = [],
  events,
  getEvents,
  activeItem,
  attendEvent,
  handleItemClick,
}) => {


  return (
    <StyledEvents>
      <Grid className='app-container'>
        <Grid.Column width={4}>
          <Menu fluid vertical tabular>
            {
              Object.keys(labelVerbMapping).map(item => (
                <Menu.Item
                  name={item}
                  active={activeItem === item}
                  onClick={handleItemClick}
                />
              ))
            }
          </Menu>
        </Grid.Column>

        <Grid.Column stretched width={12}>
          <Segment>
            <EventsSection key="user-upcoming-events-section"
                           title={activeItem} {...{events, getEvents, attendEvent }} />,
          </Segment>
        </Grid.Column>
      </Grid>
    </StyledEvents>
  )
}

export default MyEvents