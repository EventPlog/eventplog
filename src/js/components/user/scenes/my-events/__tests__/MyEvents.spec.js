import React from 'react';
import MyEvents from '../MyEvents';
import EventsSection from 'js/components/events/scenes/events/components/events-section'
import { shallow } from 'enzyme';

describe('MyEvents', () => {
  it('should render correctly', () => {
    const wrapper = shallow( <MyEvents  /> );

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(EventsSection).length).toEqual(1);
  })
});