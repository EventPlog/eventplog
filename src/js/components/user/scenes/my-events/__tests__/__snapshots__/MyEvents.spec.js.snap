// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyEvents should render correctly 1`] = `
<MyEvents__StyledEvents>
  <Grid
    className="app-container"
  >
    <GridColumn
      width={4}
    >
      <Menu
        fluid={true}
        tabular={true}
        vertical={true}
      />
    </GridColumn>
    <GridColumn
      stretched={true}
      width={12}
    >
      <Segment>
        <EventsSection
          key="user-upcoming-events-section"
        />
      </Segment>
    </GridColumn>
  </Grid>
</MyEvents__StyledEvents>
`;
